tests/conftest.py

```diff
@@ -37,21 +35,27 @@ def spark(app_name="Sample", url="local[*]", memory="1G"):
         .config("spark.executor.instances", 1)
         .config("spark.executor.heartbeatInterval", "36000s")
         .config("spark.network.timeout", "10000000s")
-        .config("spark.driver.maxResultSize", memory)
-    """
-    SUBMIT_ARGS = "--packages eisber:sarplus:0.2.3 pyspark-shell"
-    os.environ["PYSPARK_SUBMIT_ARGS"] = SUBMIT_ARGS
 
+    Args:
+        app_name (str): sets name of the application
+        url (str): url for spark master
+
+    Returns:
+        SparkSession: new Spark session
+    """
+    n_cores = get_number_processors()
+    physical_mem = get_physical_memory()
     return (
         SparkSession.builder.appName(app_name)
         .master(url)
-        .config("spark.driver.memory", memory)
-        .config("spark.sql.shuffle.partitions", "1")
+        .config("spark.driver.cores", 1)
+        .config("spark.driver.maxResultSize", "1g")
+        .config("spark.driver.memory", "{:.2f}g".format(physical_mem * 0.2))
+        .config("spark.executor.cores", n_cores - 1)
+        .config("spark.executor.instances", 1)
+        .config("spark.executor.memory", "{:.2f}g".format(physical_mem * 0.6))
         .config("spark.local.dir", "/mnt")
-        .config("spark.worker.cleanup.enabled", "true")
-        .config("spark.worker.cleanup.appDataTtl", "3600")
-        .config("spark.worker.cleanup.interval", "300")
-        .config("spark.storage.cleanupFilesAfterExecutorExit", "true")
+        .config("spark.sql.shuffle.partitions", 1)
```

Comment:
I think the error comes because  the value is a string:

("spark.sql.shuffle.partitions","n")

Reply:
From the Azure DevOps logs, it looked like the JVM didn't like fractional memory amounts. I've fixed this now; let's see if it runs...